frb_codegen/src/parser/mod.rs

```diff
@@ -488,3 +560,60 @@ fn extract_metadata(attrs: &[Attribute]) -> Vec<IrDartAnnotation> {
 fn type_to_string(ty: &Type) -> String {
     quote!(#ty).to_string().replace(' ', "")
 }
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashMap;
+
+    use syn::{parse_str, Type};
+
+    use crate::parser::topo_resolve;
+
+    #[test]
+    fn test_topo_resolve1() {
+        let mut input = HashMap::new();
+        let mut expect = HashMap::new();
+        input.insert("id".to_string(), parse_str::<Type>("i32").unwrap());
```

Comment:
uber-nit: easier hashmap initialization: https://stackoverflow.com/a/58126168/4619958 (btw I wrote the answer ;) )